src/components/engagement/engagement.service.ts

```diff
@@ -312,6 +315,10 @@ export class EngagementService {
       input.pnp,
```

Comment:
Why doesn't `createDefinedFile` check for `input.pnp`?

Reply:
Since all defined files could be nullable we just save the if statement everywhere and do it inside the function.